### PR TITLE
feat: add capability for navigation to display unit based on back key

### PR DESCRIPTION
**Pull request by ts-nitish-a-jain**
# Description
Adds the capability to go backward onBackKeyPressed

## Links
[MINI-1094](https://jira.rakuten-it.com/jira/browse/MINI-1094)

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

**Comment by ts-nitish-a-jain** [comment]
> Looks good 👍
> 
> But let's not merge this until after releasing v1.1.0. I'll mark this as "Request changes" for now so that it doesn't get merged on accident.

@corycaywood Thanks. It makes sense.

**Comment by ts-nitish-a-jain** [comment]
> Because we support navigation on phone back button now so I think of adding `close` button in sample app 🤔

@minh-rakuten for the close button & more functions on navigation, we would need to check with Clayton, I created the PR out of a feasibility-check getting converted into code-implementation.

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package com.rakuten.tech.mobile.miniapp
 
+import android.util.Log
```
Comment (ts-nitish-a-jain) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt:3]: @minh-rakuten I think the branch that you baselined it with is a different one.

Comment (minh-rakuten) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/MiniAppDownloader.kt]: Oops, sorry 🙇 . The same baseline but forgot to remove logger codes.
Removed and totally checked.

**Pull request merged**